discv5/discv5-theory.md

```diff
@@ -92,141 +92,199 @@ from the `k` closest nodes it has seen.
+The topic advertisement subsystem indexes participants by their provided services. A
+node's provided services are identified by arbitrary strings called 'topics'. A node
+providing a certain service is said to 'place an ad' for itself when it makes itself
+discoverable under that topic. Depending on the needs of the application, a node can
+advertise multiple topics or no topics at all. Every node participating in the discovery
+protocol acts as an advertisement medium, meaning that it accepts topic ads from other
+nodes and later returns them to nodes searching for the same topic.
+
+### Topic Table
+
+Nodes store ads for any number of topics and a limited number of ads for each topic. The
+data structure holding advertisements is called the 'topic table'. The list of ads for a
+particular topic is called the 'topic queue' because it functions like a FIFO queue of
+limited length. The image below depicts a topic table containing three queues. The queue
+for topic `T₁` is at capacity.
+
+![topic table](./img/topic-queue-diagram.png)
+
+The queue size limit is implementation-defined. Implementations should place a global
+limit on the number of ads in the topic table regardless of the topic queue which contains
+them. Reasonable limits are 100 ads per queue and 50000 ads across all queues. Since ENRs
+are at most 300 bytes in size, these limits ensure that a full topic table consumes
+approximately 15MB of memory.
+
+Any node may appear at most once in any topic queue, that is, registration of a node which
+is already registered for a given topic fails. Implementations may impose other
+restrictions on the table, such as restrictions on the number of IP-addresses in a certain
+range or number of occurrences of the same node across queues.
+
+### Tickets
+
+Ads should remain in the queue for a constant amount of time, the `target-ad-lifetime`. To
+maintain this guarantee, new registrations are throttled and registrants must wait for a
+certain amount of time before they are admitted. When a node attempts to place an ad, it
+receives a 'ticket' which tells them how long they must wait before they will be accepted.
+It is up to the registrant node to keep the ticket and present it to the advertisement
+medium when the waiting time has elapsed.
+
+The waiting time constant is:
+
+    target-ad-lifetime = 15min
+
+The assigned waiting time for any registration attempt is determined according to the
+following rules:
+
+- When the table is full, the waiting time is assigned based on the lifetime of the oldest
+  ad across the whole table, i.e. the registrant must wait for a table slot to become
+  available.
+- When the topic queue is full, the waiting time depends on the lifetime of the oldest ad
+  in the queue. The assigned time is `target-ad-lifetime - oldest-ad-lifetime` in this
+  case.
+- Otherwise the ad may be placed immediately.
+
+Tickets are opaque objects, not unlike HTTP session cookies. While details of encoding and
```

Comment:
Http session cookies are, by definition, lookup keys to a server-side session. 
You maybe mean some other type of cookie, where the state itself is encoded into the cookie? That's what I understand from context further down